rancher2/config.go

```diff
@@ -735,8 +736,9 @@ func (c *Config) WaitForClusterState(clusterID, state string, interval time.Dura
 				if obj.Conditions[i].Status == "True" {
 					return obj, nil
 				}
-				// When condition is false, checking if the cluster is transitioning to still wait for condition
-				if obj.Transitioning == "yes" {
+				// When cluster condition is false, retrying if it has been updated for last rancher2WaitFalseCond seconds
+				lasUpdate, err := time.Parse(time.RFC3339, obj.Conditions[i].LastUpdateTime)
+				if err == nil && int(time.Since(lasUpdate)/time.Second) < rancher2WaitFalseCond {
```

Comment:
nit: typo

```suggestion
				lastUpdate, err := time.Parse(time.RFC3339, obj.Conditions[i].LastUpdateTime)
				if err == nil && int(time.Since(lastUpdate)/time.Second) < rancher2WaitFalseCond {
```

Reply:
Alternatively, could we modify this logic slightly? I believe you need to cast to an int since `rancher2WaitFalseCond` is an int, but would the following work instead?

```suggestion
				lastUpdate, err := time.Parse(time.RFC3339, obj.Conditions[i].LastUpdateTime)
				if err == nil && time.Since(lastUpdate) < rancher2WaitFalseCond * time.Second {
```

Reply:
Updated variable name and style suggestion